src/components/transactions/Switch/cowprotocol.helpers.ts

```diff
@@ -0,0 +1,370 @@
+export const COW_PROTOCOL_ETH_FLOW_ADDRESS = '0xbA3cB449bD2B4ADddBc894D8697F5170800EAdeC';
+const COW_CREATE_ORDER_ABI =
+  'function createOrder((address,address,uint256,uint256,bytes32,uint256,uint32,bool,int64)) returns (bytes32)';
+
+// Until CoW shares a more sophisticated way to recognize token groups, we'll maintain a lists with popular tokens
+const TOKEN_GROUPS: Record<'stable' | 'correlatedEth' | 'correlatedBtc', string[]> = {
+  stable: [
+    'USDC',
+    'USDT',
+    'DAI',
+    'GHO',
+    'EURC',
+    'USDbC',
+    'USDe',
+    'USDS',
+    'sUSDe',
+    'RLUSD',
+    'PYUSD',
+    'LUSD',
+    'sDAI',
+    'crvUSD',
+    'USD₮0',
+    'USDC.e',
+    'EURe',
+    'XDAI',
+  ],
+  correlatedEth: [
+    'weETH',
+    'ETH',
+    'WETH',
+    'wstETH',
+    'cbETH',
+    'ezETH',
+    'wrsETH',
+    'osETH',
+    'rETH',
+    'ETHx',
+  ],
+  correlatedBtc: ['cbBTC', 'WBTC', 'LBTC', 'tBTC', 'eBTC'],
+} as const;
+
+const cowSymbolGroup = (symbol: string): keyof typeof TOKEN_GROUPS | 'unknown' => {
+  for (const [groupName, tokens] of Object.entries(TOKEN_GROUPS)) {
+    if (tokens.includes(symbol)) {
+      return groupName as keyof typeof TOKEN_GROUPS;
+    }
+  }
+  return 'unknown';
+};
+
+export const HEADER_WIDGET_APP_CODE = 'aave-v3-interface-widget';
+export const ADAPTER_APP_CODE = 'aave-v3-interface-aps'; // Use this one for contract adapters so we have different dashboards
+export const COW_PARTNER_FEE: (tokenFromSymbol: string, tokenToSymbol: string) => PartnerFee = (
+  tokenFromSymbol: string,
+  tokenToSymbol: string
+) => ({
+  bps: cowSymbolGroup(tokenFromSymbol) == cowSymbolGroup(tokenToSymbol) ? 15 : 25,
+  recipient: COW_EVM_RECIPIENT,
+});
+export const COW_APP_DATA: (tokenFromSymbol: string, tokenToSymbol: string) => AppDataRootSchema = (
+  tokenFromSymbol: string,
+  tokenToSymbol: string
+) => {
+  return {
+    appCode: HEADER_WIDGET_APP_CODE, // todo: use ADAPTER_APP_CODE for contract adapters
+    version: '1.3.0',
+    metadata: {
+      partnerFee: COW_PARTNER_FEE(tokenFromSymbol, tokenToSymbol),
+    },
+  };
+};
+
+export type CowProtocolActionParams = {
+  quote: OrderParameters;
+  provider: JsonRpcProvider;
+  chainId: number;
+  user: string;
+  amount: string;
+  destAmount: string;
+  tokenDest: string;
+  tokenSrc: string;
+  tokenSrcDecimals: number;
+  tokenDestDecimals: number;
+  inputSymbol: string;
+  outputSymbol: string;
+};
+
+export const getPreSignTransaction = async ({
+  provider,
+  tokenDest,
+  chainId,
+  user,
+  amount,
+  tokenSrc,
+  tokenSrcDecimals,
+  tokenDestDecimals,
+  destAmount,
+  inputSymbol,
+  outputSymbol,
+}: CowProtocolActionParams) => {
+  if (!isChainIdSupportedByCoWProtocol(chainId)) {
+    throw new Error('Chain not supported.');
+  }
+
+  const signer = provider?.getSigner();
+  if (!signer) {
+    throw new Error('No signer found in provider');
+  }
+
+  const tradingSdk = new TradingSdk({ chainId, signer, appCode: HEADER_WIDGET_APP_CODE });
```

Comment:
nit: This is fine, but please note that you can have a single SDK instance if you wish, and then pass the signer, and chainId in the methods. It will merge the constructor params with what you pass later.